src/quicktype-core/language/CPlusPlus.ts

```diff
@@ -1067,15 +1295,19 @@ export class CPlusPlusRenderer extends ConvenienceRenderer {
                 if (typeForKind === undefined) continue;
                 this.emitLine(onFirst ? "if" : "else if", " (_j.", func, "())");
                 this.indent(() => {
-                    this.emitLine(
-                        "_x = _j.get<",
-                        this.cppType(
-                            typeForKind,
-                            { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: true },
-                            false
-                        ),
-                        ">();"
+                    const cppType = this.cppType(
+                        typeForKind,
+                        { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: true },
+                        false,
+                        true
                     );
+                    let EncodeValueStart: string = "";
+                    let EncodeValueEnd: string = "";
+                    if (this._options.wstring && cppType === "std::string") {
```

Comment:
This is repeated so many times.  I'd be very much obliged if you abstracted this whole string literal construction into an interface with two implementations: one for regular and one for wide strings.  I can do that, too, if you can't be bothered - let me know.